### DISCUSSION TITLE
Do someone have any ideas that can set wallpapers on Niri ? Possibly using 'feh' or 'swww'

### DISCUSSION THREAD

**Discussion by gentrywang** [YaLTeR/niri#879] [Q&A]
I just switch WM from sway to Niri, I have no idea how to set the wallpapers on Niri, Please offer me proper advices

**Comment by YaLTeR** [comment]
Just use the same swww setup

**Comment by zetaPRIME** [comment]
yeah, waypaper remains great as always

**Comment by cyrneko** [comment]
pretty easy

if you wanna do swww, simply install it according to swww's own instructions, add `spawn-at-startup "swww-daemon"` somewhere in your niri config (probably somewhere at the top) and...that's it! swww should now work and you can set a wallpaper by running `swww img /path/to/image`. Swww will remember your wallpaper, too.

**Reply by Akirapearl** [reply]
i tried multiple options, even a local shell script, can you disclose a bit further what worked for you??, my current attempt it this:

```
// NEW CONFIGS

// Set wallpaper, first needs for swww to be installed (best option I found so far,
// since swaybg works as a command but not as a configuration like spawn-at-startup)
spawn-at-startup "./scripts/wallp.sh"
```

Content of wallp.sh
```
#!/bin/bash/

swww-daemon &
sleep 0.5
swww img /home/$USER/Pictures/thumb-1920-1377649.png

```

* $USER is a replacement of my current username

**Reply by cyrneko** [reply]
doing the above is unnecessary, SWWW remembers which wallpaper to use for which output automatically. Just running `swww img /path` once is enough, no need to run it every time Niri starts.

**Comment by alicealysia** [comment]
swww with waypaper-engine so you have a gui to adjust your backgrounds

**Comment by Sempyos** [comment]
I just use matugen + swww and then have a plugin for the file manager Yazi that lets me set my wallpapers from there.
